lib/rust/profiling/src/lib.rs

```diff
@@ -349,12 +365,12 @@ impl IntervalHandle {
 
 impl Drop for IntervalHandle {
     fn drop(&mut self) {
-        if !self.released {
-            warn_on_error(mark_end_interval(self.metadata.clone()));
-            WARNING!(format!(
-                "{} was dropped without explicitly being ended.",
-                self.metadata.label
-            ));
+        if self.released {
+            return;
+        }
+        if let Some(metadata) = self.metadata.take() {
+            WARNING!(format!("{} is dropped without explicitly being ended.", &metadata.label));
+            warn_on_error(mark_end_interval(metadata, self.stats_guard.take()));
         }
```

Comment:
Why we dont warning about dropping the handle without explicit end if the metadata was None? What is the reasoning behind it? I see 2 possibilities – no metadata might mean that the profiler is not started or something – and then this design is too implicit and very confusing. Another possibility is just a mistake in the code. Nevertheless, this code should be fixed, because it does not make any sense to the reader that if there is no metadata, this thing can be dropped without a warning.